### ISSUE TITLE
SQLCipher 4.1.1 - UnsatisfiedLinkError

### ISSUE THREAD

**Issue by iskugor** [sqlcipher/android-database-sqlcipher#433]
### Expected Behavior

App should not crash.

### Actual Behavior

```
Caused by: java.lang.UnsatisfiedLinkError: dlopen failed: cannot locate symbol "__aeabi_memcpy4" referenced by "/data/app/com.commonsware.cwac.saferoom.test-2/lib/arm/libsqlcipher.so"...
at java.lang.Runtime.loadLibrary(Runtime.java:372)
at java.lang.System.loadLibrary(System.java:1076)
at net.sqlcipher.database.SQLiteDatabase$1.loadLibraries(SQLiteDatabase.java:223)
at net.sqlcipher.database.SQLiteDatabase.loadLibs(SQLiteDatabase.java:240)
at net.sqlcipher.database.SQLiteDatabase.loadLibs(SQLiteDatabase.java:219)
at net.sqlcipher.database.SQLiteDatabase.loadLibs(SQLiteDatabase.java:212)
```

### Steps to Reproduce

You can see this issue by running android tests from: https://github.com/commonsguy/cwac-saferoom

You just need to upgrade min sdk version in saferoom library.

I could see the issue just on Android 6 (on Samsung A3 device).

SQLCipher version (can be identified by executing `PRAGMA cipher_version;`):

4.1.1

SQLCipher for Android version:

4.1.1

**Comment by commonsguy** [comment]
SafeRoom may not be set up yet for SQLCipher for Android 4.1.1. That's a project for my upcoming weekend. I recommend that you stick with 4.0.1 until such time as I release a new SafeRoom version.

**Comment by iskugor** [comment]
@commonsguy thanks for your answer! Yes, I plan to stick to 4.0.1.
However, I think this is issue with Sqlcipher no matter SafeRoom (as I said, it only happens on Samsung device with Android 6), I just used SafeRoom test case just so I don't have to create my own test app. :)

**Comment by developernotes** [comment]
Hi @iskugor 

Would you try running the SQLCipher for Android [test suite](https://github.com/sqlcipher/sqlcipher-android-tests) on that device an report your results?

**Comment by iskugor** [comment]
@developernotes yes, I'm having same crash

`
Caused by: java.lang.UnsatisfiedLinkError: dlopen failed: cannot locate symbol "__aeabi_memcpy4" referenced by "/data/app/net.zetetic.sqlcipher.test-1/lib/arm/libsqlcipher.so"...
        at java.lang.Runtime.loadLibrary(Runtime.java:372)
        at java.lang.System.loadLibrary(System.java:1076)
        at net.sqlcipher.database.SQLiteDatabase$1.loadLibraries(SQLiteDatabase.java:223)
        at net.sqlcipher.database.SQLiteDatabase.loadLibs(SQLiteDatabase.java:240)
        at net.sqlcipher.database.SQLiteDatabase.loadLibs(SQLiteDatabase.java:219)
        at net.sqlcipher.database.SQLiteDatabase.loadLibs(SQLiteDatabase.java:212)
        at net.zetetic.tests.TestSuiteRunner.runSuite(TestSuiteRunner.java:53)
        at net.zetetic.tests.TestSuiteRunner.doInBackground(TestSuiteRunner.java:34)
        at net.zetetic.tests.TestSuiteRunner.doInBackground(TestSuiteRunner.java:17)
        at android.os.AsyncTask$2.call(AsyncTask.java:295)
        at java.util.concurrent.FutureTask.run(FutureTask.java:237)
        at android.os.AsyncTask$SerialExecutor$1.run(AsyncTask.java:234) 
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1113) 
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:588) 
        at java.lang.Thread.run(Thread.java:818) 
`

**Comment by developernotes** [comment]
Hi @iskugor 

We recently [release](https://discuss.zetetic.net/t/sqlcipher-for-android-4-1-2-release/3515) SQLCipher for Android 4.1.2 which adjusts our API support back down to 14 on 32-but CPU's. Would you give that a try and let us know your results? Thanks!

**Comment by TheNephilim88** [comment]
Even 4.1.2 does not work for me, UnsatisfiedLinkError too. E.g. with Emulator API 21 x86_64 platform.

**Comment by developernotes** [comment]
Hi @TheNephilim88 

Thank you for the report, we are able to reproduce this issue within an emulator and will investigate further.

**Comment by brodybits** [comment]
I think the following line should be fixed: <https://github.com/sqlcipher/android-database-sqlcipher/blob/master/android-database-sqlcipher/native.gradle#L128>

It should use the min sdk value setting.

P.S. I found the following issue that may be related, from a quick search: <https://github.com/android-ndk/ndk/issues/126>

**Comment by developernotes** [comment]
Hi @brodybits 

Yes, that is correct. We have a fix locally for this, but are doing some additional testing before releasing. Thanks!

**Comment by developernotes** [comment]
Hello @iskugor, @TheNephilim88, @brodybits 

We have recently published SQLCipher for Android 4.1.3 which should address the crashing behavior you experienced on the x86_64 emulator running API 21. It should be available via Maven Central within the next hour or so. Would you try out 4.1.3 when you get the opportunity and let us know your results? Thanks!

**Comment by iskugor** [comment]
@developernotes it's fixed in 4.1.3, thanks.

**Comment by developernotes** [comment]
Hi @iskugor 

Great, thanks for trying out the latest release. I will close this ticket now, please feel free to open another ticket if you run into an issue. Thanks!
